### ISSUE TITLE
Try training the model on multi-gpus, but failed.

### ISSUE THREAD

**Issue by luxuriance19** [WojciechMormul/multigpu-cnn#1]
Hello:
 I try to train the model using multi-gpus, but the code turn out to be training just on one card. There is no warning and error. The code is attached. Is there anything wrong? Thanks, Sorry about the format.

`
        tower_grads = []
            with tf.variable_scope(tf.get_variable_scope()):
                for i in range(config.num_gpus):
                    with tf.device('/gpu:%d'%i):
                        print('/gpu:%d'%i)
                        with tf.name_scope('%s_%d' % ('tower', i)) as scope:
                            ''' Dequeues one batch for the GPU'''
                            self.images, self.labels = batch_queue.dequeue()
                            
                            loss = self.tower_loss(scope, images_shape, labels_shape)
                            

                            ''' Reuse variables for the next tower.'''
                            tf.get_variable_scope().reuse_variables()
                            '''Retain the summaries from the final tower.'''
                            summaries = tf.get_collection(tf.GraphKeys.SUMMARIES, scope)
                            tvars = tf.trainable_variables()
                            '''Calculate the gradients for the batch of data on this tower.'''
                            grads = optimizer.compute_gradients(loss, tvars)
                            tower_grads.append(grads)

            print_num_of_total_parameters(output_to_logging=True)
            grads = self.average_gradients(tower_grads)
            summaries.append(tf.summary.scalar("learning_rate", self.learning_rate))
`
`
    def tower_loss(self, scope, images_shape, labels_shape):
        """
         Calculate the total loss on a single tower running the model

        scope: unique prefix string identifying the model tower, e.g. 'tower_0'
        images_shape: 4-D tensor shape of images, [batches, H, W, C]
        labels_shape: 4-D tensor shape of labels, [batches, H, W, C]

        returns:
           Tensor of shape [] containint the total lloss for a batch of data
        """
        self.build_model(images_shape, labels_shape)
        _ = self._loss()
        losses = tf.get_collection('losses', scope)
        total_loss = tf.add_n(losses, name="total_loss")
        
        for l in losses+[total_loss]:
            loss_name = re.sub('%s_[0-9]*/'%'tower', '', l.op.name)
            tf.summary.scalar(loss_name, l)
        return total_loss
`
